wandb/sdk/wandb_artifacts.py

```diff
@@ -209,6 +213,40 @@ def add_reference(self, uri, name=None, checksum=True, max_objects=None):
         for entry in manifest_entries:
             self._manifest.add_entry(entry)
 
+    # TODO: name this add_obj?
+    def add(self, obj, name):
```

Comment:
Or `add_media`

Reply:
Just throwing something out here: Reading this class, i can foresee `Artifact#add_file` being moved to a private method, and instead, introducing a `File(Media)` class which is a general-purpose file data_type. This would make `Artifact#add` the only exposed method (moreover, we could overload the stub so that a string is treated as a file). From a rendering perspective, File could be a download link. Anyhow, that would reduce the surface area to just 1 function and possibly help reduce ambiguity.